backend/ethereum/channel/funder.go

```diff
@@ -349,6 +379,9 @@ func FundingIDs(channelID channel.ID, participants ...perunwallet.Address) [][32
 
 // NumTX returns how many Transactions are needed for the funding request.
 func (f *Funder) NumTX(req channel.FundingReq) (sum uint32, err error) {
+	f.mtx.Lock()
```

Comment:
You could use a read-write lock and allows concurrent reads. I think only `RegisterAsset` writes and should not be run concurrently. The other methods are probably fine running concurrently?

Reply:
Yes, we could also use a read write lock here. I will update the PR.

Reply:
Updated in this [force-push](https://github.com/hyperledger-labs/go-perun/compare/c5255fbe5f5453796953eb1c9e559a9565a7c2ea..4a12ab6d8062a3296b14d92a697dc705aec0b7e5).